src/cswinrt/strings/WinRT.cs

```diff
@@ -502,32 +580,117 @@ public unsafe ObjectReference<I> _ActivateInstance<I>()
             }
         }
 
-#if NET
-        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2091:RequiresUnreferencedCode",
-            Justification = "No members of the generic type are dynamically accessed in this code path.")]
-#endif
-        public ObjectReference<I> _As<I>() => _IActivationFactory.As<I>();
-        public IObjectReference _As(Guid iid) => _IActivationFactory.As<WinRT.Interop.IUnknownVftbl>(iid);
+        public ObjectReference<I> _As<I>() => Value.As<I>();
+        public IObjectReference _As(Guid iid) => Value.As<WinRT.Interop.IUnknownVftbl>(iid);
     }
 
     internal sealed class ActivationFactory<T> : BaseActivationFactory
-    {
-        public ActivationFactory() : base(typeof(T).Namespace, typeof(T).FullName) { }
+    {
+        private static readonly ActivationFactory<T> _instance = new ActivationFactory<T>();
+
+        public ActivationFactory()
+            : base(typeof(T).Namespace, typeof(T).FullName)
+        {
+        }
 
-        static readonly ActivationFactory<T> _factory = new ActivationFactory<T>();
-        public static new I AsInterface<I>() => _factory.Value.AsInterface<I>();
-        public static ObjectReference<I> As<I>() => _factory._As<I>();
-        public static IObjectReference As(Guid iid) => _factory._As(iid);
+        public static new I AsInterface<I>() => _instance.Value.AsInterface<I>();
+        public static ObjectReference<I> As<I>() => _instance._As<I>();
+        public static IObjectReference As(Guid iid) => _instance._As(iid);
 
-#if NET
-        [UnconditionalSuppressMessage("ReflectionAnalysis", "IL2091:RequiresUnreferencedCode", 
-            Justification = "No members of the generic type are dynamically accessed in this code path.")]
-#endif
         public static ObjectReference<I> ActivateInstance<
 #if NET
             [DynamicallyAccessedMembers(DynamicallyAccessedMemberTypes.None)]
```

Comment:
Is this actually needed? Isn't `I` already not annotated for any dynamic access in this context? 🤔